### PR TITLE
gcl update

### PR DESCRIPTION
**Pull request by g-e-o**

### CODE REVIEW

File: asm/libgcl/desktop.ini [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[LocalizedFileNames]
```
Comment (paulsapps) [inline asm/libgcl/desktop.ini:1]: you can prob delete this file ;)

File: build/linker_command_file.txt [hunk 2]
```diff
@@ -395,21 +395,22 @@
...
-	include	"..\obj\sub_80020EAC.obj" ; GCL_SetLoadFile ?
-	include	"..\obj\sub_80021264.obj" ; GCL_InitVar ?
-	include	"..\obj\sub_800212CC.obj" ; GCL_InitClearVar ?
-	include	"..\obj\sub_80021314.obj" ; GCL_SaveVar ?
-	include	"..\obj\sub_80021488.obj" ; GCL_RestoreVar ?
-	include	"..\obj\sub_80021634.obj" ; GCL_GetVar?
-	include	"..\obj\sub_8002171C.obj" ; GCL_SetVar? 
-	include	"..\obj\sub_800217F0.obj" ; GCL_VarSaveBuffer ?
+	include	"..\obj\libgcl\GCL_VarSave_80020B90.obj"
+	include	"..\obj\CRC32_80020BB4.obj"
```
Comment (paulsapps) [inline build/linker_command_file.txt:403]: is CRC32 probably part of libgcl then?

File: build/statcov.txt [hunk 3]
```diff
@@ -61,7 +61,7 @@ Functions never called at all:
     96: ..\obj\System_void_allocation_80016078.obj: function System_void_allocation_80016078
```
Comment (paulsapps) [inline build/statcov.txt:61]: prob don't need statcov.txt either, can prob add it to .gitignore ?

File: src/Game/item_kill_80033F88.c [hunk 4]
```diff
@@ -24,11 +24,11 @@ void item_kill_80033F88(int param_1)
     {
         if ((int)pbVar1 < 0)
         {
-            GCL_Run_80020118(pbVar1, (void *)0x0);
+            GCL_ExecBlock_80020118(pbVar1, (void *)0x0);
```
Comment (paulsapps) [inline src/Game/item_kill_80033F88.c:27]: you can just use NULL instead of casting 0x0, an odd Ghidra-ism

File: src/Game/item_kill_80033F88.c [hunk 5]
```diff
@@ -24,11 +24,11 @@ void item_kill_80033F88(int param_1)
...
         }
         else 
         {
-            GCL_RunOrCancelProc_8001FF2C((unsigned int)pbVar1, (void *)0x0);
+            GCL_ExecProc_8001FF2C((unsigned int)pbVar1, (void *)0x0);
```
Comment (paulsapps) [inline src/Game/item_kill_80033F88.c:31]: ditto

File: src/libgcl/GCL_GetVar_80021634.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+unsigned char *GCL_GetVar_80021634(unsigned char *pScript, int *retCode, int *retValue)
+{
+    int		gcl_var;
+    int		gcl_code;
+    char	*ptr;
+
+    gcl_var = GCL_GetLong(pScript);
+    gcl_code = GCL_GetVarTypeCode(gcl_var);
+    *retCode = gcl_code;
+	if (GCL_IsGameStateVar(gcl_var))
```
Comment (paulsapps) [inline src/libgcl/GCL_GetVar_80021634.c:27]: tab

File: src/libgcl/GCL_GetVar_80021634.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+//
+// gcl_variable example: 0x11800002 (difficulty_level)
+//     1 = type variable
+//     1 = type short
+//    80 = gameState struct
+//  0002 = offset in gameState struct
+
+unsigned char *GCL_GetVar_80021634(unsigned char *pScript, int *retCode, int *retValue)
+{
+    int		gcl_var;
```
Comment (paulsapps) [inline src/libgcl/GCL_GetVar_80021634.c:20]: mix of tabs and spaces in this one

File: src/libgcl/gcl.h [hunk 8]
```diff
@@ -92,34 +93,67 @@ enum GCLOperators
...
 }
 
-#define GCL_ReadShort(p)    ( p[1] ) | ( p[0] << 8 )
 #define GCL_AdvanceShort(p) p += sizeof(short)
-
-#define GCL_ReadByte(p)     p[0]
 #define GCL_AdvanceByte(p)  p += sizeof(unsigned char)
 
 
+void						GCL_StartDaemon_8001FCDC(void);
```
Comment (paulsapps) [inline src/libgcl/gcl.h:145]: tabbed

File: asm/libgcl/desktop.ini (hunk 1 above)
Comment (g-e-o) [reply asm/libgcl/desktop.ini:1]: I don't have this file locally

File: src/libgv/GV_ZeroMemory_8001619c.c [hunk 9]
```diff
@@ -1,31 +1,33 @@
...
-    for (iterator = lengthToClear / (sizeof(DWORD)*4); iterator > 0; iterator--)
-    {
-        *(pMemToClearWorkingCopy) = 0;
-        *(pMemToClearWorkingCopy + 1) = 0;
-        *(pMemToClearWorkingCopy + 2) = 0;
-        *(pMemToClearWorkingCopy + 3) = 0;
-        pMemToClearWorkingCopy += 4;
-    }
-    pMemToClear = (BYTE*)pMemToClearWorkingCopy;
+void            GV_ZeroMemory_8001619c( to, size )
```
Comment (paulsapps) [inline src/libgv/GV_ZeroMemory_8001619c.c:5]: wow this is............... ⚰️

File: build/linker_command_file.txt (hunk 2 above)
Comment (g-e-o) [reply build/linker_command_file.txt:403]: I'm not entirely sure

File: asm/libgcl/desktop.ini (hunk 1 above)
Comment (paulsapps) [reply asm/libgcl/desktop.ini:1]: its marked as hidden, try show hidden files in windows explorer, maybe desktop.ini is another for .gitignore file

**Pull request merged**